lib/nn.js

```diff
@@ -1,144 +1,208 @@
 // Other techniques for learning
 
-class ActivationFunction{
-  constructor(func, dfunc){
-    this.func = func;
-    this.dfunc = dfunc;
-  }
+class ActivationFunction {
+    constructor(func, dfunc, use_X_values = false) {
+        this.func = func;
+        this.dfunc = dfunc;
+        this.use_X_values = use_X_values;
+    }
 }
 
+// Range (0, 1)
 let sigmoid = new ActivationFunction(
-  x => 1 / (1 + Math.exp(-x)),
-  y => y * (1- y)
+    x => 1 / (1 + Math.exp(-x)),
+    y => y * (1 - y)
 );
 
+// Range (-1, 1)
 let tanh = new ActivationFunction(
-  x => Math.tanh(x),
-  y => 1-(y*y)
+    x => Math.tanh(x),
+    y => 1 - (y * y)
+);
+
+// Range (-PI/2, PI/2)
+let arctan = new ActivationFunction(
+    x => Math.atan(x),
+    y => 1 / (y * y) + 1,
+    use_X_values = true
+);
+
+// Range (-1, 1)
+let softsign = new ActivationFunction(
+    x => x / 1 + Math.abs(x),
```

Comment:
`x => x / 1 + Math.abs(x),` should be `x => x / (1 + Math.abs(x)),` right?

Reply:
Gosh darn it man. You are a genius.